exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -49,6 +49,7 @@
   String USER_RPC_TIMEOUT = "drill.exec.rpc.user.timeout";
   String METRICS_CONTEXT_NAME = "drill.exec.metrics.context";
   String USE_IP_ADDRESS = "drill.exec.rpc.use.ip";
+  String BIT_ADVERTISED_HOST = "drill.exec.rpc.bit.advertised.host";
```

Comment:
For an example, see `LogsResources.java` in package `org.apache.drill.exec.server.rest`:

    private File getLogFolder() {
      return new File(Preconditions.checkNotNull(System.getenv("DRILL_LOG_DIR"), "DRILL_LOG_DIR variable is not set"));
    }

The above example hard-codes the env var name. For now, you can probably do the same. When you reference the env var, maybe say something like:

    // Use a custom host name. See drill-env.sh for details.

    ... System.getenv( "DRILL_HOST_NAME" ) ...

Then, in drill-env.sh, add a commented-out entry:

    # Provide a customized host name for when the default mechanism
    # is not accurate.

    # export DRILL_HOST_NAME=`hostname`

I suspect that should be plenty good to document the purpose.

With this, I think you can omit the config variable as Drill's config system doesn't seem to have the right semantics for a per-host setting.